RFC-0001-row-id.md

```diff
@@ -0,0 +1,80 @@
+# **RFC 1 for Presto**
+
+## Row ID
+
+Proposers
+
+* Brian Dahmen
+* Devang Shah
+* Elliotte Rusty Harold
+* Serge Druzkin
+* Sreeni Viswanadha
+
+## Related Issues
+
+* Row ID support #22078
+* Binary IDs for Hive partitions #22103
+* Binary rowID type #22053
+* Start adding row\_id hidden column to Hive #22008
+
+
+## Summary
+
+A row ID is a binary key for a row that is globally unique across the warehouse.
```

Comment:
By "warehouse", do we mean data lake?

Reply:
I find the definition of [data lake](https://en.wikipedia.org/wiki/Data_lake) on Wikipedia more relevant for this proposal than [data warehouse](https://en.wikipedia.org/wiki/Data_warehouse).

Reply:
Per wikipedia "A data lake is a system or [repository of data](https://en.wikipedia.org/wiki/Data_repository) stored in its natural/raw format,[[1]](https://en.wikipedia.org/wiki/Data_lake#cite_note-1) usually object [blobs](https://en.wikipedia.org/wiki/Binary_large_object) or files. A data lake is usually a single store of data including raw copies of source system data, sensor data, social data etc.,[[2]](https://en.wikipedia.org/wiki/Data_lake#cite_note-2)"

Since that includes a lot of non-tabular data, I don't think it describes what Presto operates on.

Reply:
Presto not operating over data lakes is news to lots of people. :)  I don't mean to be nit-picky, but I think outside of Meta, when you use the term Warehouse, it typically means Enterprise Data Warehouse--which are data appliances that include a front (engine) and back (storage) component.  Presto can operate over that, but this proposal talks about identifying row IDs over files and partitions on distributed storage that is cataloged by an HMS or similar system, which is a lake.  Lakes often include unstructured data in addition to structured files, and there are other systems that use this parallel to Presto.  This flexibility in the structure of the data is what makes them appealing, and doesn't have any bearing on Presto's ability over to operate over the lake.